parser-typechecker/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -1362,34 +1375,36 @@ showDiffNamespace ppe d@OBD.BranchDiffOutput{..} =
   prettyAddTypes :: [OBD.AddedTypeDisplay v a] -> Numbered Pretty
   prettyAddTypes = fmap P.lines . traverse prettyGroup where
     prettyGroup :: OBD.AddedTypeDisplay v a -> Numbered Pretty
-    prettyGroup (hqmds, _r, odecl) = do
-      pairs <- traverse (prettyLine odecl) hqmds
+    prettyGroup (hqmds, r, odecl) = do
+      pairs <- traverse (prettyLine r odecl) hqmds
       let (nums, decls) = unzip pairs
       let boxLeft = case hqmds of _:_:_ -> P.boxLeft; _ -> id
       pure . P.column2 $ zip nums (boxLeft decls)
-    prettyLine :: Maybe (DD.DeclOrBuiltin v a) -> (HQ'.HashQualified, [OBD.MetadataDisplay v a]) -> Numbered (Pretty, Pretty)
-    prettyLine odecl (hq, mds) = do
-      n <- num
+    prettyLine :: Reference -> Maybe (DD.DeclOrBuiltin v a) -> (HQ'.HashQualified, [OBD.MetadataDisplay v a]) -> Numbered (Pretty, Pretty)
+    prettyLine r odecl (hq, mds) = do
+      n <- numHQ newPath hq (Referent.Ref r)
       pure . (n,) $ prettyDecl hq odecl <> case length mds of
         0 -> mempty
-        c -> "(+" <> P.shown c <> " metadata)"
+        c -> " (+" <> P.shown c <> " metadata)"
 
   prettyAddTerms :: [OBD.AddedTermDisplay v a] -> Numbered Pretty
   prettyAddTerms = fmap (P.column3 . mconcat) . traverse prettyGroup where
     prettyGroup :: OBD.AddedTermDisplay v a -> Numbered [(Pretty, Pretty, Pretty)]
-    prettyGroup (hqmds, _r, otype) = do
-      pairs <- traverse (prettyLine otype) hqmds
+    prettyGroup (hqmds, r, otype) = do
+      pairs <- traverse (prettyLine r otype) hqmds
       let (nums, names, decls) = unzip3 pairs
           boxLeft = case hqmds of _:_:_ -> P.boxLeft; _ -> id
       pure $ zip3 nums (boxLeft names) decls
-    prettyLine otype (hq, mds) = do
-      n <- num
-      pure . (n, phq' hq, ) $ " : " <> prettyType otype
+    prettyLine r otype (hq, mds) = do
+      n <- numHQ newPath hq r
+      pure . (n, phq' hq, ) $ " : " <> prettyType otype <> case length mds of
+        0 -> mempty
+        c -> " (+" <> P.shown c <> " metadata)"
 
   prettySummarizePatch, prettyNamePatch :: OBD.PatchDisplay -> Numbered Pretty
   --  12. patch p (added 3 updates, deleted 1)
   prettySummarizePatch (name, patchDiff) = do
-    n <- num
+    n <- numPatch name
```

Comment:
Todo: Whoops, `numPatch` needs to take a `Path.Absolute` prefix too, like `numHQ`.